test/unit/org/apache/cassandra/cql3/ReservedKeywordsTest.java

```diff
@@ -30,14 +34,51 @@ public void testReservedWordsForColumns()
     {
         for (String reservedWord : ReservedKeywords.reservedKeywords)
         {
-            try
-            {
-                QueryProcessor.parseStatement(String.format("ALTER TABLE ks.t ADD %s TEXT", reservedWord));
+            if (isAllowed(reservedWord))
                 Assert.fail(String.format("Reserved keyword %s should not have parsed", reservedWord));
-            }
-            catch (SyntaxException ignore)
-            {
-            }
+        }
+    }
+
+    @Test
+    public void parserAndTextFileMatch()
```

Comment:
this test isn't 100% replacing the other one, as there is always the chance the Lexer is missing something in the text file, so the other one is still useful